src/scripts.rs

```diff
@@ -446,6 +446,27 @@ impl DerivedDepositDescriptor {
 
         Ok(DerivedDepositDescriptor(deposit_desc!(stakeholders)))
     }
+
+    /// Get all the keys and key source used in this derived Deposit descriptor
+    pub fn keys(&self) -> Vec<DerivedPublicKey> {
```

Comment:
Can't you put this method, DerivedUnvaultDescriptor::keys and DerivedCpfpDescriptor::keys directly into `impl_descriptor_newtype`, in the `derived_struct_name`? Or there are some differences between the three I'm not seeing?

Reply:
Good call. Done in a new commit to move both `xpubs()` and `keys()`